### PR TITLE
add user guide about the code migration

### PR DESCRIPTION
**Pull request by XuhuiRen**
Signed-off-by: XuhuiRen <xuhui.ren@intel.com>

## Type of Change

Add the introduction about the migration from Intel Neural Compressor 1.X to 2.X

## Description

[detail description 
JIRA ticket: xxx](https://jira.devtools.intel.com/browse/ILITV-2621)

### CODE REVIEW

**Comment by chensuyue** [comment]
Do we need a detail `yaml -> config class` table for each API?

**Comment by XuhuiRen** [comment]
> All done. Hope for more suggestions.

File: docs/source/migration.md [hunk 1]
```diff
@@ -0,0 +1,716 @@
...
+              momentum: 0.9
+              weight_decay: 0.0004
+              nesterov: False
+    evaluation:                              # optional. required if user doesn't provide eval_func in neural_compressor.Quantization.
+    accuracy:                              # optional. required if user doesn't provide eval_func in neural_compressor.Quantization.
+        metric:
+        topk: 1                            # built-in metrics are topk, map, f1, allow user to register new metric.
+        dataloader:
+        batch_size: 256
+        dataset:
```
Comment (NeoZhangJianyu) [inline docs/source/migration.md:398]: how to implement the 'dataloader' item in 2.x API?
Expect to see how to replace every items in yaml by the codes of 2.x API.

File: docs/source/migration.md [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+> It should be noted that we will stop to support Intel Neural Compressor 1.X in the future.
+
+1. [Quantization](#model-quantization)
+2. [Pruning](#pruning)
+3. [Distillation](#distillation)
+4. [Mix-Precision](#mix-precision)
+5. [Orchestration](#orchestration)
+6. [Benchmark](#benchmark)
+7. [Examples](#examples)
+
```
Comment (NeoZhangJianyu) [inline docs/source/migration.md:15]: For advanced user, how to find the map rule for the parameters in yaml which are not listed in this guide?
Is there any common method to fix it?
for example:
  neural_compressor.config.PostTrainingQuantConfig  <-> quantization in yaml file.

Comment (XuhuiRen) [reply docs/source/migration.md:15]: The parameters that are listed in this guide do not influence the actual tuning performance. I followed the previous yaml template to map from 1.x to 2.x. In the configuration of 2.x, I have introduced the mapping relation one by one. It is very easy for the user to find the desired name of the parameters.

File: docs/source/migration.md (hunk 1 above)
Comment (XuhuiRen) [reply docs/source/migration.md:398]: There is a lot of configuration information in the previous yaml file do not have any function. Not all parameters have their own mapping pattern. We have deleted the unnecessary items in 2.x API. We directly input the dataloader to the train function. I will add a sentence about it. The other information is no more used.

File: docs/source/user_guide.md [hunk 3]
```diff
@@ -1,3 +1,90 @@
...
+  <tbody>
+      <tr>
+          <td colspan="1" align="center"><a href="./docs/source/adaptor.md">Adaptor</a></td>
+          <td colspan="2" align="center"><a href="./docs/source/tuning_strategies.md">Strategy</a></td>
+          <td colspan="3" align="center"><a href="./docs/source/distillation_quantization.md">Distillation for Quantization</a></td>
+          <td colspan="3" align="center"><a href="./docs/source/metric.md">Metric</a></td>    
+      </tr>
+      <tr>     
+        <td colspan="3" align="center"><a href="./docs/source/objective.md">Objective</a></td>
+        <td colspan="3" align="center"><a href="./docs/source/migration.md">Code Migration</a></td>
```
Comment (chensuyue) [inline docs/source/user_guide.md:85]: `Code Migration` should be highlight in the very beginning, not in this corner. And it's not an `advanced topic` it should belong to `basic topic`.

**Review by NeoZhangJianyu — APPROVED** [review]
Approve

**Review by xin3he — APPROVED** [review]

**Pull request merged**